src/realm/object-store/shared_realm.cpp

```diff
@@ -467,7 +467,14 @@ void Realm::update_schema(Schema schema, uint64_t version, MigrationFunction mig
             std::swap(m_schema, schema);
             std::swap(m_schema_version, temp_version);
         });
-        initialization_function(shared_from_this());
+
+        try {
+            initialization_function(shared_from_this());
```

Comment:
This code implicitly assumes that all the SDKs will throw an exception that C++ runtime understands if something bad happened. For the ones using the C API, `CallbackFailed` is thrown in case of an error. JS and Cocoa should be using something that is either C++ or compatible with C++.